### ISSUE TITLE
About SNP Encoding and CNN in Cropformer

### ISSUE THREAD

**Issue by li-dongxu** [jiekesen/Cropformer#3]
Dear authors,

Thank you for your creative and inspiring work on **Cropformer**! I have a couple of questions regarding the encoding method and model architecture used in your paper:

1. I noticed that for encoding the genotypes (SNPs), you assigned specific values based on different nucleotide pairs as follows:
`"AA": 0, "AT": 1, "TA": 1, "AC": 2, "CA": 2, "AG": 3, "GA": 3, "TT": 4, "TC": 5, "CT": 5, "TG": 6, "GT": 6, "CC": 7, "CG": 8, "GC": 8, "GG": 9`.
Then, you normalized these values and made them into 10000-d tensors. Could you please elaborate on the biological reasoning behind this choice? Additionally, have you considered alternative approaches like using `nn.Embedding()` instead of direct numerical normalization?

2. In Figure 1B of your paper, you illustrated the CNN layer with a label `Convolution 3×3`; however, in the code, I noticed that only a `Conv1d()` operation is used with a kernel size of 3. Could you clarify if this is a discrepancy or if I might have misunderstood? Also, could you elaborate on the biological significance of this convolution operation in the context of SNP data?

Thank you again for your excellent work! I would appreciate it if you could take the time to address my questions.

Best regards,
Don

**Comment by jiekesen** [comment]
Thank you for your question. We've talked to you and answered it.

**Comment by ssmfmf** [comment]
Dear author and Dear li-dongxu，
  I also have the same question regarding SNP coding. By directly converting base pairs into numerical codes, wouldn’t the model mistakenly interpret these numbers as representing ordinal values (e.g., 0 < 1 < 2)? 
Looking forward to your answers.

**Comment by li-dongxu** [comment]
@ssmfmf Well you are right, and the author told me that this is THEIR way of representing the genotypes, while I totally disagree with this approach. Numerical relation should not be posed when all the ATGC are equally important in single nucleotide variation.
